### ISSUE TITLE
Better debug log format

### ISSUE THREAD

**Issue by burner1024** [phobos2077/sfall#195]
I'm currently looking at this:

![captura de pantalla de 2018-09-21 02-09-54](https://user-images.githubusercontent.com/1279852/45841281-7f3d9180-bd43-11e8-8bb4-fc1506d15c84.png)

Which lead me to think, maybe log format can be improved? Force line breaks, at least.
Prefixing messages with actual script name and timestamp also could be useful.

**Comment by FakelsHub** [comment]
это лог не от sfall, его непосредственно создает движок игры.
и переписывать косяковые строки в движке... ну у меня как бы есть более важные задачи)

**Comment by FakelsHub** [comment]
Если ты поспособствуешь и выпишешь строки где нет перевода сток которые тебе встречаются, то я могу тебе скриптик для игры накатать который будет фиксить в экзешнике отсутствующий символ перевода строки.

**Comment by burner1024** [comment]
я понимаю что не от sfall... переписывать руками мне кажется неоптимально. Может, можно вхучится в фунцию лога собственно? Или ее нет единой?
Так-то руками захачить я и сам смогу наверное.

**Comment by FakelsHub** [comment]
единая конечно.
те ты перелагаешь проверять строку на наличие символа перевода строки?

а всякие префиксы не ну его нафиг)

**Comment by burner1024** [comment]
Да, умнее было бы проверять... если только отсутствие перевода не используется где-то специально, чтобы кусками выводить строку.

**Comment by phobos2077** [comment]
Если правильно помню, за вывод в лог отвечает встроенная функция которая по всей видимости просто выводит строку в файл, без переноса на новую строку или имени скрипта. Вроде была еще одна функция более высокого уровня которая добавляла имя скрипта и т.п. Sfall вроде бы её как раз и юзает. Чтобы это исправить надо пройтись по всему коду и заменять вызовы первой функции на вторую, НО только там где это имеет смысл. Кароче гемор. Можно попытаться впихнуть в строку ошибки перенос, но оно того не стоит по-моему.

**Comment by FakelsHub** [comment]
>  если только отсутствие перевода не используется где-то специально

Во-во.

**Comment by burner1024** [comment]
Is it possible to at least add script name to "Obj is null" and "var out of range" messages? That would allow to fish them out and fix one by one eventually.
I'm aware `HideObjIsNullMsg` was added recently, but hiding is not quite the same as fixing.

**Comment by FakelsHub** [comment]
"Obj is null" - не является ошибкой, поэтому скрываем как излишняя спам инфа, все остальное надо исправлять, а не скрывать.

**Comment by burner1024** [comment]
Ну я то и говорю.

**Comment by FakelsHub** [comment]
А что ты говоришь?, на английском мне не понятна суть)

**Comment by FakelsHub** [comment]
А понял.
